pkg/scheduler/algorithmprovider/defaults/register_predicates.go

```diff
@@ -103,20 +103,11 @@ func init() {
 	// (e.g. kubelet and all schedulers)
 	factory.RegisterFitPredicate(predicates.GeneralPred, predicates.GeneralPredicates)
 
-	// Fit is determined by node memory pressure condition.
-	factory.RegisterFitPredicate(predicates.CheckNodeMemoryPressurePred, predicates.CheckNodeMemoryPressurePredicate)
```

Comment:
If we remove this, it would break backward compatibility. https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/82703/pull-kubernetes-bazel-test/1174606131515887619

/cc @ahg-g

Reply:
The latest commit https://github.com/kubernetes/kubernetes/pull/82703/commits/f642bc2feb755cb6f834787163725a498cda4dce fix the failing test

Reply:
This predicate was effectively removed throughout beta, so it seems the compatibility test didn't take into account feature gates? in any case, I think we should keep them registered to maintain backward compatibility, add a comment to mark them as deprecated and just remove them from the default list.

Reply:
But we have to register `CheckNodeConditionPred` as a mandatory fit predicate and remove it when apply the feature gates

Reply:
does CheckNodeConditionPred have to be registered as mandatory to keep compatibility? I think doing ```RegisterFitPredicate(predicates.CheckNodeConditionPred, predicates.CheckNodeConditionPredicate)``` would be enough.

Reply:
> does CheckNodeConditionPred have to be registered as mandatory to keep compatibility? I think doing `RegisterFitPredicate(predicates.CheckNodeConditionPred, predicates.CheckNodeConditionPredicate)` would be enough.

What about the latter one?

```go
--- FAIL: TestCompatibility_v1_Scheduler (0.01s)
    compatibility_test.go:1216: 1.16: Expected:
        	api.Policy{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, Predicates:[]api.PredicatePolicy{api.PredicatePolicy{Name:"MatchNodeSelector", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"PodFitsResources", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"PodFitsHostPorts", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"HostName", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"NoDiskConflict", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"NoVolumeZoneConflict", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"PodToleratesNodeTaints", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"MaxEBSVolumeCount", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"MaxGCEPDVolumeCount", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"MaxAzureDiskVolumeCount", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"MaxCSIVolumeCountPred", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"MaxCinderVolumeCount", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"MatchInterPodAffinity", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"GeneralPredicates", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"CheckVolumeBinding", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"CheckNodeUnschedulable", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"TestServiceAffinity", Argument:(*api.PredicateArgument)(0xc0002ca7c0)}, api.PredicatePolicy{Name:"TestLabelsPresence", Argument:(*api.PredicateArgument)(0xc0002ca7e0)}}, Priorities:[]api.PriorityPolicy{api.PriorityPolicy{Name:"EqualPriority", Weight:2, Argument:(*api.PriorityArgument)(nil)}, api.PriorityPolicy{Name:"ImageLocalityPriority", Weight:2, Argument:(*api.PriorityArgument)(nil)}, api.PriorityPolicy{Name:"LeastRequestedPriority", Weight:2, Argument:(*api.PriorityArgument)(nil)}, api.PriorityPolicy{Name:"BalancedResourceAllocation", Weight:2, Argument:(*api.PriorityArgument)(nil)}, api.PriorityPolicy{Name:"SelectorSpreadPriority", Weight:2, Argument:(*api.PriorityArgument)(nil)}, api.PriorityPolicy{Name:"NodePreferAvoidPodsPriority", Weight:2, Argument:(*api.PriorityArgument)(nil)}, api.PriorityPolicy{Name:"NodeAffinityPriority", Weight:2, Argument:(*api.PriorityArgument)(nil)}, api.PriorityPolicy{Name:"TaintTolerationPriority", Weight:2, Argument:(*api.PriorityArgument)(nil)}, api.PriorityPolicy{Name:"InterPodAffinityPriority", Weight:2, Argument:(*api.PriorityArgument)(nil)}, api.PriorityPolicy{Name:"MostRequestedPriority", Weight:2, Argument:(*api.PriorityArgument)(nil)}, api.PriorityPolicy{Name:"RequestedToCapacityRatioPriority", Weight:2, Argument:(*api.PriorityArgument)(0xc000179840)}}, ExtenderConfigs:[]api.ExtenderConfig{api.ExtenderConfig{URLPrefix:"/prefix", FilterVerb:"filter", PreemptVerb:"", PrioritizeVerb:"prioritize", Weight:1, BindVerb:"bind", EnableHTTPS:true, TLSConfig:(*api.ExtenderTLSConfig)(0xc000154b40), HTTPTimeout:1, NodeCacheCapable:true, ManagedResources:[]api.ExtenderManagedResource{api.ExtenderManagedResource{Name:"example.com/foo", IgnoredByScheduler:true}}, Ignorable:true}}, HardPodAffinitySymmetricWeight:0, AlwaysCheckAllPredicates:false}
        Got:
        	api.Policy{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, Predicates:[]api.PredicatePolicy{api.PredicatePolicy{Name:"MatchNodeSelector", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"PodFitsResources", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"PodFitsHostPorts", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"HostName", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"NoDiskConflict", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"NoVolumeZoneConflict", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"PodToleratesNodeTaints", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"CheckNodeMemoryPressure", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"CheckNodeDiskPressure", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"CheckNodePIDPressure", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"CheckNodeCondition", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"MaxEBSVolumeCount", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"MaxGCEPDVolumeCount", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"MaxAzureDiskVolumeCount", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"MaxCSIVolumeCountPred", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"MaxCinderVolumeCount", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"MatchInterPodAffinity", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"GeneralPredicates", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"CheckVolumeBinding", Argument:(*api.PredicateArgument)(nil)}, api.PredicatePolicy{Name:"TestServiceAffinity", Argument:(*api.PredicateArgument)(0xc00045fc30)}, api.PredicatePolicy{Name:"TestLabelsPresence", Argument:(*api.PredicateArgument)(0xc00045fc60)}}, Priorities:[]api.PriorityPolicy{api.PriorityPolicy{Name:"EqualPriority", Weight:2, Argument:(*api.PriorityArgument)(nil)}, api.PriorityPolicy{Name:"ImageLocalityPriority", Weight:2, Argument:(*api.PriorityArgument)(nil)}, api.PriorityPolicy{Name:"LeastRequestedPriority", Weight:2, Argument:(*api.PriorityArgument)(nil)}, api.PriorityPolicy{Name:"BalancedResourceAllocation", Weight:2, Argument:(*api.PriorityArgument)(nil)}, api.PriorityPolicy{Name:"SelectorSpreadPriority", Weight:2, Argument:(*api.PriorityArgument)(nil)}, api.PriorityPolicy{Name:"NodePreferAvoidPodsPriority", Weight:2, Argument:(*api.PriorityArgument)(nil)}, api.PriorityPolicy{Name:"NodeAffinityPriority", Weight:2, Argument:(*api.PriorityArgument)(nil)}, api.PriorityPolicy{Name:"TaintTolerationPriority", Weight:2, Argument:(*api.PriorityArgument)(nil)}, api.PriorityPolicy{Name:"InterPodAffinityPriority", Weight:2, Argument:(*api.PriorityArgument)(nil)}, api.PriorityPolicy{Name:"MostRequestedPriority", Weight:2, Argument:(*api.PriorityArgument)(nil)}, api.PriorityPolicy{Name:"RequestedToCapacityRatioPriority", Weight:2, Argument:(*api.PriorityArgument)(0xc000518ca0)}}, ExtenderConfigs:[]api.ExtenderConfig{api.ExtenderConfig{URLPrefix:"/prefix", FilterVerb:"filter", PreemptVerb:"", PrioritizeVerb:"prioritize", Weight:1, BindVerb:"bind", EnableHTTPS:true, TLSConfig:(*api.ExtenderTLSConfig)(0xc0005cd710), HTTPTimeout:1, NodeCacheCapable:true, ManagedResources:[]api.ExtenderManagedResource{api.ExtenderManagedResource{Name:"example.com/foo", IgnoredByScheduler:true}}, Ignorable:true}}, HardPodAffinitySymmetricWeight:0, AlwaysCheckAllPredicates:false}
    compatibility_test.go:1252: Registered predicates are missing from compatibility test (add to test stanza for version currently in development): []string{"CheckNodeUnschedulable"}
```